include/solvers/postprocessing_velocities.h

```diff
@@ -0,0 +1,156 @@
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+#include <deal.II/lac/trilinos_vector.h>
+
+// Dofs
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+// Numerics
+#include <deal.II/numerics/solution_transfer.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// Lethe Includes
+#include <core/parameters.h>
+#include <core/simulation_control.h>
+#include "navier_stokes_solver_parameters.h"
+#include "post_processors.h"
+
+
+using namespace dealii;
+
+
+/**
+ * @brief AverageVelocities. The AverageVelocities class calculates the
+ * time-averaged velocities and pressure. The generated vector is output
+ * with the solution and visualization is possible.
+ */
+template <int dim, typename VectorType, typename DofsType>
+class AverageVelocities
+{
+public:
+  void
+  calculate_average_velocities(
+    const VectorType &                       local_evaluation_point,
+    const std::shared_ptr<SimulationControl> &simulation_control,
+    const Parameters::PostProcessing &       post_processing,
+    const DofsType &                         locally_owned_dofs,
+    const MPI_Comm &                         mpi_communicator);
+
+  const VectorType
+  get_average_velocities();
+
+  const VectorType
+  get_average_velocities(const double bulk_velocity);
+
+
+private:
+  TrilinosScalar inv_range_time;
+  TrilinosScalar dt;
+
+  VectorType sum_velocity_dt;
+  VectorType average_velocities;
+};
+
+/**
+ * @brief calculate_average_velocities. This function calculates time-averaged
+ * velocities and pressure with dof vector with no ghost cell.
+ *
+ * @param local_evaluation_point. The vector solutions with no ghost cells
+ *
+ * @param simulation_control. The simulation information (time)
+ *
+ * @param post_processing. The parameters to start the processing
+ *
+ * @param locally_owned_dofs. The owned dofs
+ *
+ * @param mpi_communicator. The mpi communicator information
+ */
+template <int dim, typename VectorType, typename DofsType>
+void
+AverageVelocities<dim, VectorType, DofsType>::calculate_average_velocities(
+  const VectorType &                       local_evaluation_point,
+  const std::shared_ptr<SimulationControl> &simulation_control,
+  const Parameters::PostProcessing &       post_processing,
+  const DofsType &                         locally_owned_dofs,
+  const MPI_Comm &                         mpi_communicator)
+{
+  double time = simulation_control->get_current_time();
+  double total_time = time - post_processing.initial_time;
+  dt = simulation_control->calculate_time_step();
+
+  if (simulation_control->get_step_number() == 0)
+  {
+    // Reinitializing vectors with zeros and dt at t = 0
+    sum_velocity_dt.reinit(locally_owned_dofs,
+                           mpi_communicator);
+    average_velocities.reinit(locally_owned_dofs,
+                              mpi_communicator);
+  }
+  else if (abs(total_time) < 1e-6 || total_time > 0)
```

Comment:
I'm not sure I understand || here? If total_time >0 it should be sufficient, why the abs check?
Time is also always positive no?

Reply:
The condition is once the initial time is reached do ... 
But with doubles I'm never sure that my condition is gonna be good if I don't use a tolerance because total_time is a subtraction. There's absolutely no chance total_time is going to be below 0? Or I guess if I test it and it works, it should works everytime.